Bug 1830095: Update Control Plane status logic to include no. of not healthy compo…

rawagner opened the pull request:
…ments

![Screenshot from 2020-05-07 20-33-46](https://user-images.githubusercontent.com/2078045/81331751-585fda80-90a2-11ea-974f-8ebd7fb369eb.png)
![Screenshot from 2020-05-07 20-34-12](https://user-images.githubusercontent.com/2078045/81331757-5b5acb00-90a2-11ea-9713-264f85ae1d89.png)

@andybraren 
If data fetching from Prometheus failed we show `x not available`
If there are no data -> `x unknown`
If the success rate is low -> `x degraded`

Does that make sense ? Do you think its worth having not available and unknown status here ? We already have that for Operators health.

andybraren commented:
I think those conditions for `not available`, `unknown`, and `degraded` make sense, and having them be separate things. 👍

One small detail if the layout works okay would be writing `1 component degraded/unknown/not available` since:

```
Control Plane
1 degraded
```

kind of sounds like 1 Control Plane is degraded, which is a little odd. "Operators" being plural makes the wording there a little different.

rawagner commented:
![Screenshot from 2020-05-11 09-30-57](https://user-images.githubusercontent.com/2078045/81535614-69db0800-936a-11ea-9ecf-b73e4ef923a2.png)
![Screenshot from 2020-05-11 09-30-23](https://user-images.githubusercontent.com/2078045/81535618-6a739e80-936a-11ea-892d-b3f2a9002634.png)

Updated to include `components` in status detail
`1 component unknown` / `x components unknown`

I've also updated Operators status detail message to be in lower case - before `1 Degraded`, after `1 degraded`

mareklibra commented:
/lgtm
